test/functional/apps/dashboard/group5/dashboard_panel_listing.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['dashboard', 'header', 'visualize', 'settings', 'common']);
+  const testSubjects = getService('testSubjects');
+  const kibanaServer = getService('kibanaServer');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+
+  describe('dashboard panel listing', () => {
+    before(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.importExport.load(
+        'test/functional/fixtures/kbn_archiver/dashboard/current/kibana'
+      );
+      await kibanaServer.uiSettings.replace({
+        defaultIndex: '0bf35f60-3dc9-11e8-8660-4d65aa086b3c',
+      });
+    });
+
+    after(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+    });
+
+    it('renders a panel with predefined order of panel groups', async () => {
+      await PageObjects.dashboard.navigateToApp();
+      await PageObjects.dashboard.clickNewDashboard();
+      await PageObjects.dashboard.switchToEditMode();
+
+      await dashboardAddPanel.clickEditorMenuButton();
+
+      const panelSelectionList = await testSubjects.find('dashboardPanelSelectionList');
+
+      const panelGroupByOrder = new Map();
+
+      const panelGroups = await panelSelectionList.findAllByCssSelector(
+        '[data-test-subj*="dashboardEditorMenu-"]'
+      );
+
+      for (let i = 0; i < panelGroups.length; i++) {
+        const panelGroup = panelGroups[i];
+        const order = await panelGroup.getAttribute('data-group-sort-order');
+        // @ts-ignore
+        const [, panelGroupTitle] = (await panelGroup.getAttribute('data-test-subj'))?.match(
+          /dashboardEditorMenu-(.*)/
+        );
+
+        panelGroupByOrder.set(order, panelGroupTitle);
+      }
+
+      expect([...panelGroupByOrder.values()]).to.eql([
+        'visualizationsGroup',
+        'annotation-and-navigationGroup',
+        'observabilityGroup',
+        'legacyGroup',
+      ]);
+    });
```

Comment:
> I wonder if it would be simpler + quicker to just check that `panelSelectionList` has 4 children? If this is just meant as a smoke test, I think that would be enough.

It could be, my reasoning here is we are able to ascertain that the panels are rendered in a specific order by reading the test assertion. For context this whole PR came out of a discussion where o11y added a new group, also had this new group displayed above annotations, in that instance we would know that there's an extra plan but not necessarily it's order.

Reply:
That's a fair point! The order **is** probably something we want to enforce, since stuff like Lens **should** remain at the top  :) I'm good with this!